core/src/main/java/org/opensearch/sql/datasource/DataSourceService.java

```diff
@@ -19,7 +19,7 @@ public interface DataSourceService {
    *
    * @return set of {@link DataSource}.
    */
-  Set<DataSource> getDataSources();
+  Set<DataSourceMetadata> getMaskedDataSourceMetadataSet();
```

Comment:
what is masked? could you change the java doc also?

Reply:
Credential info will be removed from the metadata, so masked. Will change the java document.

Reply:
why change to return DataSourceMetadata, not sure the reason, but I think DataSourceService has getDataSources method is more natural?

Reply:
Let me cover all the use cases in description and will see if it is required or will revert. 
Show datasources command works with DataSourceService which actually requires datasource metadata but not datasource. 
But with the current information that we provide we can work with getDatasources as well.